Automattic/woocommerce-payments
# Sepa/1539 pay for order

Fixes #1539 
Fixes #1522 

#### Changes proposed in this Pull Request

Populate customer details from order or from customer when on "Pay for Order" or "Add Payment Method".

This also fixes a bug where the owner name would be set to the string 'undefined undefined'.

Note:
Payments will **not** go through on Pay for Order when using CC if on the order the name is left blank.
Payments will **not** go through on Pay for Order when using SEPA if on the order the name and billing email are left blank.

Add Payment method for SEPA is not working yet. It will be done on #1538 









#### Testing instructions





* Set store currency to euro
* Enable feature flag wp option, update `_wcpay_feature_sepa` to 1
* Enable SEPA in checkbox at bottom of `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments`
* Create a manual order for a customer or guest
* Make sure to enter name and email on billing details.
* Log out and visit the Pay for Order link as the customer or guest
* Select SEPA Method. Enter the test number and submit.
* Verify the payment goes through, verify the customer details in Stripe are correct.
* Verify the customer name is correct on the site transactions page.
* Repeat with CC.

-------------------
